strings/longest_substr_without_repeating_chars.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * @file
+ * @brief Greedy algorithm to find length of longest substring without repeating
+ * characters
+ *
+ * @details
+ * We are given a string. We need to compute the length of the longest
+ * substring, such that no character in the substring occurs more than once.
+ *
+ * For a string of length n, the time complexity of this algorithm is O(n)
+ * (amortized)
+ *
+ * @author [Nishant Mahesh](https://github.com/nishant-mahesh)
+ */
+
+#include <algorithm>  // To use the inbuilt 'max' function
+#include <cassert>
+#include <iostream>
+#include <map>
+#include <string>
+
+using namespace std;
```

Comment:
This is not good practice. It is best to use `std::` directly.
Or you can do this instead. This is good practice.

```suggestion
using std::cout;
using std::cin;
using std::string;
```

You can add more using sentences as well. 🙂